apps/librelingo_json_export/tests/test_challenges_clean_word.py

```diff
@@ -0,0 +1,30 @@
+from utils import FakeLanguages
+from librelingo_utils import clean_word
+
+
+def test_empty_string():
+    assert clean_word("") == ""
+
+
+def test_removes_parentheses():
+    assert clean_word("(foo") == FakeLanguages.LANG_1
```

Comment:
this is confusing

1. why extract this value to a constant/class? I don't see the point in that as this function basically works on generic strings. 
2. `FakeLanguages.LANG_1` sounds like this is a language. But it's actually a cleaned word.

I would probably remove this `FakeLanguages` thing

Reply:
Oh you are right, I am still trying to adapt 100% with the codebase, the tests and all the workspace of the current work, i created the enum for the FakeWords "foo" "bar" as there were a lot of repetition of them just to void this string repetition (i will probably have to change the `clean_word("foo")` into a `clean_word(f"({...}")`, i missed some of the `foo` used. But why do you believe it's a bad idea to have an enum to avoid code repetition since it allow us if once we want to change all the "foo" into "foozibarzibar" to do it just once instead of having to recheck all over the codebase to find them ?

I also was confused a little bit why it uses constants words as "foo" but also not constant ints (with randint) since the tests are normally made to have exactly the same behavior even if we run them 10000 times to have a deterministic behavior with values.
While the random values may lead to a failing tests due to a edge case value that will not be reproducable or it will but with a 1/5000 probability.

Reply:
a discussion i was inspired of when i first started working on testing:

https://softwareengineering.stackexchange.com/questions/221632/testing-deterministic-or-non-deterministic#:~:text=When-,every,-run%20of%20the

https://stackoverflow.com/questions/32458/random-data-in-unit-tests

Also the "foo" "bar" ... values maybe can be more "realistic" words that can appear in the real app?

Reply:
Ok i rollbacked it to "foo" "bar" variables, but just poining out that this is what misleaded me :) 

```python
        x, y = "foo", "bar"
        z = str(fakes.fake_value())
        p1 = fakes.customize(fakes.word1, in_source_language=[x])
        p2 = fakes.customize(fakes.word1, in_source_language=[x, y])
        self.assertEqual(get_dumb_opaque_id(z, p1), get_dumb_opaque_id(z, p2))
```

Would like to have your input concerning the actual naming and repetition on variables!

<img width="333" alt="Screenshot 2022-01-07 at 04 16 02" src="https://user-images.githubusercontent.com/66479002/148485727-c6857b2b-3eef-470e-9aae-8b680bbba748.png">

Reply:
> created the enum for the FakeWords "foo" "bar" as there were a lot of repetition of them just to void this string repetition

is that type of string repetition bad? 

the main reason I prefer it this way is because this way it demonstrates the usage of the function more (no need to look up the meaning of the constant)

Reply:
> I also was confused a little bit why it uses constants words as "foo" but also not constant ints (with randint) since the tests are normally made to have exactly the same behavior even if we run them 10000 times to have a deterministic behavior with values.
While the random values may lead to a failing tests due to a edge case value that will not be reproducable or it will but with a 1/5000 probability.

this is kind of a good point, using this type of randomness is probably not the best idea

Reply:
> Also the "foo" "bar" ... values maybe can be more "realistic" words that can appear in the real app?

yes, and this would effectively be a better way to also get rid of the code duplication